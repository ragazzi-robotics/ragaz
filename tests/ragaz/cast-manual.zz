# test: {"no_auto_cast": ""}


def main():

    var a : i16 = 1
    var b : i32 = 2
    var c = a as i32 + b
    print(c)

    var d: f32 = 1.0
    var e: f32 = 2.0
    var f: f64 = (d / e) as f64
    print(f)

    var g = c as f64 * f
    print(g)

    print(g as i32)
