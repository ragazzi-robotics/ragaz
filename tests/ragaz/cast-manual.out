3
0.500000
1.500000
1
